### ISSUE TITLE
[FEAT]: Scoring System

### ISSUE THREAD

**Issue by ImBIOS** [ImBIOS/gitmemoji#1]
### Clear and concise description of the problem

It'd be nice if we can show off to our friends about our score in gitmemoji quiz.

### Validations

- [X] Follow our [Code of Conduct](https://github.com/ImBIOS/.github/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://github.com/ImBIOS/contribute).
- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

**Comment by tymnec** [comment]
I would like to add scoring system in this project! :rocket:

**Comment by tymnec** [comment]
Please assign this to me

**Comment by ImBIOS** [comment]
@tymnec Before I can assign you, We need to discuss this first. How's your idea to approach this feature request?

**Comment by ImBIOS** [comment]
I'm thinking of using streak system and use firebase to store the leaderboard.

**Comment by tymnec** [comment]
Additionally, we can add a timer to limit how long the user has to select an answer, and deduct points for incorrect answers.

**Comment by ImBIOS** [comment]
We probably need a new UI design for this. The UI I got from the past maintainer was kinda bad in the UX.

**Comment by tymnec** [comment]
Alright What do you think UI design should be? Do you have any similar design?

**Comment by ImBIOS** [comment]
None, not yet doing my own design research for it. In your opinion, how the design should be? I'm thinking of creating a Figma Design for this. Do you ever use Figma? I'd like to communicate design with it.

**Comment by tymnec** [comment]
Yup I had used Figma a lot. Just start creating design of it I'll join

**Comment by tymnec** [comment]
Also let me know the UTC time when you will start working on it.
